src/Containers/DenseAxisArray.jl

```diff
@@ -147,7 +147,17 @@ function Base.get(
 end
 
 _abstract_vector(x::AbstractVector) = x
-_abstract_vector(x) = [a for a in x]
+function _abstract_vector(x)
+    axis = [a for a in x]
+    if length(axis) == 1
+        @warn(
+            "Axis contains one element: $x. If intended, you can safely " *
+            "ignore this warning. To explicitly pass the axis with one " *
+            "element, pass `[$(axis[1])]` instead of `$(x)`.",
```

Comment:
`[...]` won't work if you pass an iterator of one element. Maybe you want to check that `x isa eltype(axis)`